cfn-resources/cloud-backup-schedule/cmd/resource/resource.go

```diff
@@ -17,193 +17,117 @@ package resource
-}
-
-// logger setup function
 func setup() {
 	util.SetupLogger("mongodb-atlas-cloud-backup-schedule")
 }
 
-// Create handles the Create event from the Cloudformation service.
 func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
-	// logger setup
 	setup()
-	_, _ = logger.Debugf("Create() currentModel:%+v", currentModel)
-
-	// Validate required fields in the request
-	if errEvent := validateModel(RequiredFields, currentModel); errEvent != nil {
-		_, _ = logger.Warnf("Validation Error")
-		return *errEvent, nil
-	}
-
-	// Validate the Profile
-	if currentModel.Profile == nil || *currentModel.Profile == "" {
-		currentModel.Profile = aws.String(profile.DefaultProfile)
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	if err := validator.ValidateModel(RequiredFields, currentModel); err != nil {
+		return *err, nil
 	}
 
-	client, pe := util.NewMongoDBClient(req, currentModel.Profile)
-	if pe != nil {
-		_, _ = logger.Warnf("CreateMongoDBClient error: %v", *pe)
-		return *pe, nil
-	}
-
-	// API call to Create cloud backup schedule
-	return cloudBackupScheduleCreateOrUpdate(req, prevModel, currentModel, client)
+	return cloudBackupScheduleCreateOrUpdate(req, prevModel, currentModel)
 }
 
-// Read handles the Read event from the Cloudformation service.
 func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
-	// logger setup
 	setup()
-	_, _ = logger.Debugf("Get the current snapshot schedule and retention settings for the cluster:%+v", currentModel.ClusterName)
-	// Validate required fields in the request
-	if errEvent := validateModel(RequiredFields, currentModel); errEvent != nil {
-		_, _ = logger.Warnf("Validation Error")
-		return *errEvent, nil
-	}
-	// Create atlas client
-	if currentModel.Profile == nil || *currentModel.Profile == "" {
-		currentModel.Profile = aws.String(profile.DefaultProfile)
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	if err := validator.ValidateModel(RequiredFields, currentModel); err != nil {
+		return *err, nil
 	}
-	client, pe := util.NewMongoDBClient(req, currentModel.Profile)
+
+	client, pe := util.NewAtlasClient(&req, currentModel.Profile)
 	if pe != nil {
-		_, _ = logger.Warnf("CreateMongoDBClient error: %v", *pe)
 		return *pe, nil
 	}
 
-	projectID := *currentModel.ProjectId
-	clusterName := *currentModel.ClusterName
-
-	// API call to Get the cloud backup schedule
-	backupPolicy, resp, err := client.CloudProviderSnapshotBackupPolicies.Get(context.Background(), projectID, clusterName)
+	backupPolicy, resp, err := client.AtlasV2.CloudBackupsApi.GetBackupSchedule(context.Background(), *currentModel.ProjectId, *currentModel.ClusterName).Execute()
 	if err != nil {
-		return progressevents.GetFailedEventByResponse(fmt.Sprintf("Error deleting cloud backup schedule : %s", err.Error()),
-			resp.Response), nil
+		return progressevent.GetFailedEventByResponse(err.Error(), resp), nil
```

Comment:
small detail, but I believe small prefix description before error can help to quickly identify from which part of the code the error was thrown

Reply:
I'm favoring to avoid specific message if they don't add value, here it fails in Delete.  but happy to reconsider and have a global approach for this